### PR TITLE
employee checkin late entry and early exit

### PR DESCRIPTION
**Pull request by yusuffGoodie**
## Is this a Feature, Chore or Bug?
- [] Feature
- [] Chore
- [] Bug

## Analysis and design (optional)
Analyse and attach the design documentation

## Solution description
Describe your code changes in detail for reviewers.

## Is there a business logic within a doctype?
    - [] Yes
    - [] No

## Output screenshots (optional)
Post the output screenshots, if a UI is affected or added due to this feature.

## Areas affected and ensured
List out the areas affected by your code changes.

## Is there any existing behavior change of other features due to this code change?
Mention Yes or No. If Yes, provide the appropriate explanation.

## Did you test with the following dataset?
- [] Existing Data
- [] New Data

## Was child table created?
    - [] is attachment required?
        did you test attachment
## Did you delete custom field?
    - [] Yes
    - [] No
        If yes, did you write a delete patch?

## Is patch required?
- [] Yes
- [] No
    ## Was the patch test?

## Which browser(s) did you use for testing?
  - [] Chrome
  - [] Safari
  - [] Firefox

### CODE REVIEW

File: one_fm/overrides/employee_checkin.py [hunk 1]
```diff
@@ -110,23 +108,21 @@ def exists_checkin(current_shift_assignment, checkin_name, log_type="IN"):
 
 	return False
 
-def after_insert_background(self):
```
Comment (NoBoneZ) [inline one_fm/overrides/employee_checkin.py:113]: This will cause an issue, please check it

self and employee_checkin

File: one_fm/utils.py [hunk 2]
```diff
@@ -3480,14 +3480,15 @@ def get_current_shift(employee):
...
         if shift[0].checkin_time > nowtime:
             minutes = int((shift[0].checkin_time - nowtime).total_seconds() / 60)
-            return {"type":"Early", "data":minutes}
+            return {"type":"Early", "data": data, "time": minutes}
         elif shift[0].checkout_time < nowtime:
             minutes = int((nowtime - shift[0].checkout_time).total_seconds() / 60)
-            return {"type":"Late", "data":minutes}
+            return {"type":"Late", "data": data, "time": minutes}
         elif shift[0].checkin_time <= nowtime <= shift[0].checkout_time:
-            return {"type":"On Time", "data":frappe.get_doc("Shift Assignment", shift[0].name)}
```
Comment (NoBoneZ) [inline one_fm/utils.py:3490]: please hope you checked other places where this was being utilized, to make sure there are no issues

File: one_fm/overrides/employee_checkin.py [hunk 3]
```diff
@@ -84,8 +82,8 @@ def before_insert(self):
 	def after_insert(self):
 		frappe.db.commit()
 		self.reload()
-		if not (self.shift_assignment and self.shift_type and self.operations_shift and self.shift_actual_start and self.shift_actual_end):
-			frappe.enqueue(after_insert_background, employee_checkin=self.name)
+		# if not (self.shift_assignment and self.shift_type and self.operations_shift and self.shift_actual_start and self.shift_actual_end):
```
Comment (pjamsheer) [inline one_fm/overrides/employee_checkin.py]: Why this is commented ?

File: one_fm/overrides/employee_checkin.py [hunk 4]
```diff
@@ -32,14 +32,12 @@ def validate(self):
 		self.validate_duplicate_log()
 		if frappe.db.get_single_value("HR and Payroll Additional Settings", 'validate_shift_permission_on_employee_checkin'):
 			try:
```
Comment (NoBoneZ) [inline one_fm/overrides/employee_checkin.py]: please log the error of this one also

**Review by NoBoneZ — APPROVED** [review]

**Pull request merged**